packages/chat/src/views/View.js

```diff
@@ -106,21 +106,38 @@ const Textarea = ({ state, dispatch }) => {
       <textarea
         className='w-full' rows='5' readOnly={!!state.loading}
         value={value(state, 'content')} onChange={change.bind(null, dispatch, 'content')}
+        onKeyDown={handleKeys.bind(null, state, dispatch)}
       />
     )
   }
   return (
     <textarea
       className='w-full' rows='5' value={value(state, 'content')} onChange={change.bind(null, dispatch, 'content')}
-      onKeyDown={e=>{handleMetaEnter(e)}}
+      onKeyDown={handleKeys.bind(null, state, dispatch)}
     />
   )
 }
 
-const handleMetaEnter = e => {
-  if(e.key === 'Enter' && (e.metaKey || e.ctrlKey)){
+const handleKeys = async (state, dispatch, e) => {
+  if (e.key === 'Enter' && (e.metaKey || e.ctrlKey)) {
     e.preventDefault()
-    e.target.form.requestSubmit()
+    await sendMessage(state, dispatch, e)
+  } else if (e.key === 'ArrowUp' && (e.metaKey || e.ctrlKey) && state && state.channels) {
+    e.preventDefault()
+    const _channels = Object.values(state.channels)
+    const _current = _channels.find(c => {
+      return state.active === c.data.name
+    })
+    const _index = (_channels.indexOf(_current) - 1 < 0 ? _channels.length - 1 : _channels.indexOf(_current) - 1)
+    await changeChannel(state, dispatch, _channels[_index])
+  } else if (e.key === 'ArrowDown' && (e.metaKey || e.ctrlKey) && state && state.channels) {
+    e.preventDefault()
+    const _channels = Object.values(state.channels)
+    const _current = _channels.find(c => {
+      return state.active === c.data.name
```

Comment:
fyi in the reduce pattern, you don't change the state directly, but you dispatch an event that changes the state.